crates/vm/src/arch/config.rs

```diff
@@ -89,11 +89,11 @@ pub struct SystemConfig {
     /// This field is skipped in serde as it's only used in execution and
     /// not needed after any serialize/deserialize.
     #[serde(skip, default = "get_default_segmentation_strategy")]
-    pub segmentation_strategy: Arc<dyn SegmentationStrategy>,
+    pub segmentation_strategy: SegmentationStrategy,
```

Comment:
We still want to allow custom user-defined `SegmentationStrategy`s here, which requires the `dyn`. If there is a part that must always be run, we should have it as a separate field?